iceoryx_posh/test/integrationtests/service_discovery/test_roudi_findservice.cpp

```diff
@@ -40,6 +40,70 @@ TEST_F(RoudiFindService_test, OfferSingleMethodServiceSingleInstance)
     ASSERT_THAT(*instanceContainer.begin(), Eq(IdString("instance1")));
 }
 
+
+///@todo #359 offereService should return bool signalling success/failure
+TEST_F(RoudiFindService_test, DISABLED_OfferServiceWithDefaultServiceDescriptionFails)
+{
+    iox::runtime::InstanceContainer instanceContainer;
+    bool isServiceOffered;
+
+    ///@todo #359 offerService should not allow invalid ServiceDescriptions
+    // isServiceOffered = senderRuntime->offerService(iox::capro::ServiceDescription());
+    this->InterOpWait();
+
+    ASSERT_EQ(false, isServiceOffered);
+}
+
+TEST_F(RoudiFindService_test, ReofferedServiceWithValidServiceDescriptionCanBeFound)
+{
+    iox::runtime::InstanceContainer instanceContainer;
+    senderRuntime->offerService({"service1", "instance1"});
+    this->InterOpWait();
+    senderRuntime->stopOfferService({"service1", "instance1"});
+    this->InterOpWait();
+    senderRuntime->offerService({"service1", "instance1"});
+    this->InterOpWait();
+
+    receiverRuntime->findService({"service1", "instance1"}, instanceContainer);
+
+    ASSERT_THAT(instanceContainer.size(), Eq(1u));
+    ASSERT_THAT(*instanceContainer.begin(), Eq(IdString("instance1")));
+}
+
+TEST_F(RoudiFindService_test, OfferExsistingServiceMultipleTimesIsRedundant)
+{
+    InstanceContainer instanceContainer;
+    senderRuntime->offerService({"service1", "instance1"});
+    this->InterOpWait();
+    senderRuntime->offerService({"service1", "instance1"});
+    this->InterOpWait();
+
+    receiverRuntime->findService({"service1", "instance1"}, instanceContainer);
+
+    ASSERT_THAT(instanceContainer.size(), Eq(1u));
+    ASSERT_THAT(*instanceContainer.begin(), Eq(IdString("instance1")));
+}
+
+///@todo #387 findService to return InstanceContainer directly
```

Comment:
Please do not add `@todo` into the code. In my opinion this is a minor change but it would maybe break the API. Therefore, I suggest the following: 
Please adapt the poshRuntime method in this pull request to:
```cpp
    cxx::expected<Error, InstanceContainer> findService(const capro::ServiceDescription& serviceDescription) noexcept;
```

Deprecate the old `findService` method like:
```cpp
    [[gnu::deprecated]] cxx::expected<Error> findService(const capro::ServiceDescription& serviceDescription,
                                     InstanceContainer& instanceContainer) noexcept;
```
and test it here.  

In my opinion an integration- or unit-test story does not only contain writing test. It also contains fixing bugs and adjust behavior if you discover them (if it is possible and feasible). If we create a new issue for every misbehavior/bug we find we have to track too much and the gain is minor since we could also fix it right away.

Reply:
@nihalchari if you need some assistance I would be happy to help you!

Reply:
okay @elfenpiff  ill do it in this PR, and yes i will ask for assistance if required 👍